hive/src/test/java/org/apache/iceberg/hive/HiveCatalogBaseTest.java

```diff
@@ -116,23 +120,28 @@ public static void stopMetastore() {
     }
   }
 
-  HiveTables tables;
+  HiveCatalog catalog;
 
   @Before
   public void createTestTable() throws Exception {
-    this.tables = new HiveTables(hiveConf);
-    tables.create(schema, partitionSpec, DB_NAME, TABLE_NAME);
+    this.catalog = new HiveCatalog(hiveConf);
+    catalog.createTable(TABLE_IDENTIFIER, schema, partitionSpec, null);
   }
 
   @After
   public void dropTestTable() throws Exception {
-    metastoreClient.dropTable(DB_NAME, TABLE_NAME);
-    tables.close();
-    this.tables = null;
+    try {
+      metastoreClient.getTable(DB_NAME, TABLE_NAME);
+      metastoreClient.dropTable(DB_NAME, TABLE_NAME);
+      this.catalog.close();
```

Comment:
Why do we close the Catalog after drop?

Reply:
I think this was to avoid exhausting handlers in the test metastore. The problem is that this is instantiating new clients instead of using the pool.